xof/xof.go

```diff
@@ -0,0 +1,86 @@
+// Package xof provides an interface for eXtendable-Output Functions.
+package xof
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/cloudflare/circl/internal/sha3"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/blake2s"
+)
+
+// XOF defines the interface to hash functions that support arbitrary-length output.
+type XOF interface {
+	// Write absorbs more data into the hash's state. It panics if called
+	// after Read.
+	io.Writer
+
+	// Read reads more output from the hash. It returns io.EOF if the limit
+	// has been reached.
+	io.Reader
+
+	// Clone returns a copy of the XOF in its current state.
+	Clone() XOF
+
+	// Reset resets the XOF to its initial state.
```

Comment:
While this is copied from the blake2b definition, I had some trouble understanding how this works for keyed ones. Maybe say "discards all data appended by Write" to clarify.

Reply:
updated now